libs/langgraph/langgraph/prebuilt/handoff.py

```diff
@@ -1,37 +1,55 @@
-from typing import Optional
-
-from langchain_core.tools import StructuredTool
-
-from langgraph.graph import GraphCommand
-
-
-class HandoffTool(StructuredTool):
-    ...
-
-
-def create_handoff_tool(
-    hand_off_to: str, name: Optional[str] = None, description: Optional[str] = None
-) -> HandoffTool:
-    """Create a tool that can hand off control to another node / agent."""
-
-    def func():
-        return f"Transferred to '{hand_off_to}'!", GraphCommand(
-            update={"active_agent": hand_off_to}
-        )
-
-    if description is None:
-        description = f"Transfer to '{hand_off_to}'. Do not ask any details."
-
-    if name is None:
-        name = hand_off_to
-
-    transfer_tool = HandoffTool.from_function(
-        func,
-        name=name,
-        description=description,
-        response_format="content_and_artifact",
-        args_schema=None,
-        infer_schema=True,
-    )
-    transfer_tool.metadata = {"hand_off_to": hand_off_to}
-    return transfer_tool
+import dataclasses
+from typing import Optional, Any
+
+
+#
+# class HandoffTool(StructuredTool):
+#     destination_node: str
+
+
+# This handoff is too limited. We want to make sure that we can update
+# graph states as part of the handoff or add a human-in-the-loop step w/ an interrupt.
+# THe primitive needs to look more like a regular tool.
+# We either push information into type annotation or we use a separate class.
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class Handoff:
+    node: str
+    parent_update: Optional[Any] = None
+    internal_update: Optional[Any] = None
```

Comment:
i don't actually understand what this one means -- the state update from the handoff will always be on the same level as the node we're handing off to

Reply:
nevermind, makes more sense now (updating the state on the level of a single react agent vs updating it on the level of the parent graph / multi-agent system)